lib/pos_blockchain.go

```diff
@@ -1192,6 +1193,29 @@ func (bav *UtxoView) hasValidBlockProposerPoS(block *MsgDeSoBlock) (_isValidBloc
 	if leaderEntryFromVotingPublicKey == nil {
 		return false, nil
 	}
+
+	// Dump some debug info on the current block's proposer and the current view's leader.
+	glog.V(2).Infof(
+		"hasValidBlockProposerPoS: Printing block proposer debug info: "+
+			"\n  Epoch Num: %d, Block View: %d, Block Height: %d, Epoch Initial View: %d, Epoch Initial Block Height: %d"+
+			"\n  Leader Idx: %d, Num Leaders: %d"+
+			"\n  Expected Leader PKID: %v, Expected Leader Voting PK: %v"+
+			"\n  Expected Leader PKID from BLS Key Lookup: %v, Expected Leader Voting PK from BLS Key Lookup: %v"+
+			"\n  Block Proposer Voting PK: %v",
+		currentEpochEntry.EpochNumber,
+		block.Header.ProposedInView,
+		block.Header.Height,
+		currentEpochEntry.InitialView,
+		currentEpochEntry.InitialBlockHeight,
+		leaderIdx,
+		len(leaders),
+		hex.EncodeToString(leaderEntry.ValidatorPKID.ToBytes()),
```

Comment:
It would probably be more useful to convert to base58? Assuming this leader hasn't had its identity swapped, its PKID = its public key. Use `lib.PkToString`

Reply:
Didn't know the base58 was the standard string encoding for PKID. I'll change to that